stdlib/public/core/SIMDVectorTypes.swift.gyb

```diff
@@ -185,6 +189,33 @@ extension SIMD${n} where Scalar: BinaryFloatingPoint {
   }
 }
 
+// SWIFT_ENABLE_TENSORFLOW
+extension SIMD${n} : AdditiveArithmetic where Scalar : FloatingPoint {}
+
+extension SIMD${n} : Differentiable
+  where Scalar : Differentiable & BinaryFloatingPoint,
+        Scalar.TangentVector : BinaryFloatingPoint {
+  public typealias TangentVector = SIMD${n}
+  public typealias AllDifferentiableVariables = SIMD${n}
+  public func tangentVector(from cotangent: TangentVector) -> TangentVector {
+    return cotangent
+  }
+}
+
+extension SIMD${n}
+  where Scalar : Differentiable & BinaryFloatingPoint,
+        Scalar.TangentVector : BinaryFloatingPoint {
+  @usableFromInline
+  internal func _vjpSubscript(index: Int)
+  -> (Scalar, (Scalar.TangentVector) -> TangentVector) {
+    return (self[index], { v in
+      var zeros = Self.zero
+      zeros[index] = Scalar(v)
```

Comment:
v is of type `Scalar.TangentVector` while `zeros` is of type `SIMD3<Scalar>`, and since the only constraint on `Scalar.TangentVector` is `Scalar.TangentVector : BinaryFloatingPoint` which `Scalar` also conforms to, I have to cast it or else I I get the error `cannot assign value of type 'Scalar.TangentVector' to type 'Scalar'`. I could add the more constraining requirement of `Scalar.TangentVector == Scalar`, but I think doing this cast is better. Let me know what you think!

Reply:
cc @rxwei regarding whether to add `Scalar.TangentVector == Scalar` constraint.
I agree that not adding the constraint is better.